Propagate pluginManager into metric reporters initialization

afedulov opened the pull request:
Naive approach to make pluginManager, which is created for the FileSystem initialization, available during the initialization of the MetricReporters ( ReportSetup.fromConfiguration method).

I do not particularly like like how it turned out, but I did not want to introduce bigger changes prior to discussing them with you. The main reason for the ugliness in changing a bunch of APIs and pulling the pluginManager through is related to the fact that FileSystems and MetricsRegistry are currently not initialized in the same place. I was not sure about the implications of pushing FileSystem initializations to the later stages, closer to the Metrics system initialization. 

Example: TaskManagerRunner initializes FileSystem in its main method, but the Metrics subsystem in the constructor.

zentol committed:
[FLINK-7727] [REST] Improve error logging in StaticFileServerHandlers

leesf committed:
[FLINK-11373] Don't cut of error message in CliFrontend

afedulov committed:
[FLINK-16222][runtime] Propagate PluginManager from entrypoints into metric reporters initialization method

pnowojski commented on flink-runtime/src/main/java/org/apache/flink/runtime/entrypoint/ClusterEntrypoint.java:
```diff
@@ -197,9 +199,11 @@ protected SecurityContext installSecurityContext(Configuration configuration) th
 		synchronized (lock) {
-			initializeServices(configuration);
+
+			//TODO: Ask why FileSystem is not initialized here too.
```
It is configured before - in `configureFileSystems()` method, which is called before `runCluster`, right?

afedulov replied:
The question why. If we see FileSystem as a service, we could put their initialization inside initializeServices. Then there will be no need to propagate pluginManager from the outside, it will be scoped to the same context.

pnowojski replied:
Again I'm not sure. Frankly I'm not very familiar with this portions of the code. One question that I would ask, is if `installSecurityContext()` can require an already initialized `FileSystem`?

If you would like to move the things around here, I would suggest to do it in separate PR in which we would need to involve Till probably.

pnowojski commented on flink-runtime/src/main/java/org/apache/flink/runtime/entrypoint/ClusterEntrypoint.java:
```diff
@@ -155,13 +156,14 @@ public void startCluster() throws ClusterEntrypointException {
 		try {
-
-			configureFileSystems(configuration);
+			//TODO: push down filesystem initialization into runCluster - initializeServices (?)
```
?

afedulov replied:
It is sort of the main question that I had - if it is OK to load FileSystems not where they are loaded now, but at the later stage, for instance inside initilizeServices.

pnowojski replied:
I don't know if it is ok to move it or not, it might be risky. Two potential issues:
- `SecurityContext` - does it affect initialisation in anyway?
- We have to be sure that we initialise `FileSystems` before we are going to need them. This might be relevant if there are services that depend on `FileSystems` - if that's the case, we would have to take that into account (correctly the initialisation)

Apart from that, it might create some unknown untested side effects that I'm simply not aware of, so I would weakly vote on leaving it as it is, unless this is causing some problems for us?

afedulov replied:
Just mild irritation of some "spaghetti"-like initialization process and increased API ugliness because of leaving the process as is. I am fine with that for now, but it seems like some refactoring might be due at a later point.

pnowojski commented on flink-runtime/src/main/java/org/apache/flink/runtime/metrics/ReporterSetup.java:
```diff
@@ -174,6 +170,13 @@ private static ReporterSetup createReporterSetup(String reporterName, MetricConf
 		return reporterArguments;
 	}
 
+	private static ReporterSetup createReporterSetup(String reporterName, MetricConfig metricConfig, MetricReporter reporter) {
```
Can you revert back the order of those methods? This is clotting the review a little bit.

afedulov replied:
Done. Not sure how this happened :)

pnowojski commented on flink-runtime/src/main/java/org/apache/flink/runtime/metrics/ReporterSetup.java:
```diff
@@ -104,14 +105,9 @@ public static ReporterSetup forReporter(String reporterName, MetricConfig metric
-	public static List<ReporterSetup> fromConfiguration(final Configuration configuration) {
+	//TODO: add forwarding method fromConfiguration(final Configuration configuration) which passes null as pluginManager?
+	//TODO: add missing javadoc, mention that pluginManager is optional
+	public static List<ReporterSetup> fromConfiguration(final Configuration configuration, final PluginManager pluginManager) {
```
is the `pluginManager` parameter used in some way?

afedulov replied:
This is the method where it will be used. Current PR is only scoped around propagating the PluginManager to this method from all of the endpoints that load metrics system.

afedulov replied:
This is the method where it will be used. Current PR is only scoped around propagating the PluginManager to this method from all of the endpoints that load metrics system.

pnowojski reviewed (changes requested):
Thanks for the heads up @afedulov, I've left couple of smaller comments. 

I guess the main code that will be actually using `PluginManager` to load a metric reporter is still missing, right?

pnowojski reviewed (approved):
Sorry for the late response. I would say that this part looks good to me. I would suggest to close this PR here and open the final PR against http://github.com/apache/flink once you implement the missing part :)

NicoK committed:
[hotfix][docs] show usage if wrong option is given

NicoK committed:
[FLINK-13793][docs] build each language in a separate subprocess

When not serving docs and just building them, e.g. via `./build_docs.sh`, we
spawn sub-processes for each language build instead of running them
single-threaded (jekyll currently does not support parallel builds).

NicoK committed:
[hotfix][docs] do not include build scripts in content generation

afedulov committed:
[hotfix][docs] Minor improvements of glossary.

This closes #9694.

wangxiyuan committed:
[FLINK-14086][core] Add OperatingArchitecture Enum

StephanEwen committed:
[FLINK-15099][runtime] Add Operator Coordinators and Events

Operator Coordinators are instances that exist once per operator. While the operators run on the TaskManagers, the
coordinator runs on the JobManager. The coordinator communicates via events with the operators, typicalls to
assign work.

The first user for those coordinators would be the new source interface.
Further users we envision are sinks (for coordinated commits of metadata), or iterations (gather progress and
steer supersteps) as well as simple approximate alignments between streams (event time alignment).

JingsongLi committed:
[FLINK-14802][orc][hive] Multi vectorized read version support for hive orc read

support vectorized read for hive 1.x.

closes #10730

rkhachatryan committed:
[hotfix][runtime,test] refactor ContinuousFileProcessingRescalingTest to prepare for ContinuousFileReaderOperator migration to Mailbox Execution model

StephanEwen committed:
[hotfix][tests] Add a TestingSchedulerFactory to help instantiating the scheduler in tests

StephanEwen committed:
[hotfix][runtime] Fix checkstyle in org.apache.flink.runtime.checkpoint (main scope)

Test scope is not included in this fix.

rkhachatryan committed:
[FLINK-12484][runtime] remove checkpoint lock *usage* from production code
After b4aa6e lock is acquired for every mailbox action if needed.

rkhachatryan committed:
[FLINK-12484][runtime] remove checkpoint lock tests
After a3b6be905f lock is not managed by operators
so the test aren't relevant anymore.

rkhachatryan committed:
[FLINK-12484][runtime] push checkpoint lock from StreamTask to SourceStream

Intermediate tasks (not source/sink) don't acquire lock anymore
(as they use single-threaded model).
SourceStreamTask still has to have the lock to conform to the public API.
MockStreamTask has the lock for compatibility with the existing tests.

mans2singh committed:
[FLINK-15181][docs] Fix minor typos and mistakes in table documentation

This closes #10521

wooplevip committed:
[FLINK-15402][csv, table] Expose disable quote character option to CSV
format

This closes #10697

sunhaibotb committed:
[FLINK-14231][task] Add the quiesce() method to ProcessingTimeService and remove TimerService::awaitPendingAfterQuiesce method

yanghua committed:
[hotfix][documentation] Fix wrong time semantics description

walterddr committed:
[FLINK-15561] Unify Kerberos credentials checking

Before, we had duplicate code in HadoopModule and YarnClusterDescriptor,
now we use the same code for both. That code is refactored to a util.

walterddr committed:
[FLINK-11589] Support security module and context discovery via ServiceLoader.

zentol committed:
[hotfix][build] Move jaxb copy into execution

This allows us to specify multiple executions that copy different file sets without having to worry what side-effects it may have.

zentol committed:
[FLINK-15672][build] Migrate to log4j2

zentol committed:
[hotfix][build] Remove unnecessary log4j files

These log4j.properties files could only be used when adding a main() method to production code,
but this could be done through a test anyway.

zentol committed:
[FLINK-15672][build][tests] Migrate test log4j configurations to log4j2

zentol committed:
[hotfix][travis] Remove outdated activation properties

zentol committed:
[FLINK-15785][travis][e2e] Rework E2E test activations

- consolidate travis activations in .travis.yml
- introduce dedicated profile for each category to reduce verbosity
- use gmavenplus-plugin to merge inclusion/exclusions across profiles
- add java11 profile (i.e., don't rely on exclusion setup in root pom)
- disable tests with Hadoop category by default
- add Hadoop activations to e2e profiles

zentol committed:
[FLINK-15685][e2e] Kafka SQL test requires Hadoop

zhengcanbin committed:
[FLINK-15646][kubernetes] Make kubernetes context configurable

This closes #10956 .

HuangZhenQiu committed:
[FLINK-12343][yarn] Add YARN file replication option

This closes #10980 .

zentol committed:
[FLINK-15672][build] Migrate production log4j configuration to log4j2

zhengcanbin committed:
[FLINK-15652][kubernetes] Support for imagePullSecrets K8s option

This closes #11054 .

rkhachatryan committed:
[FLINK-13955][runtime] use mailbox execution model in ContinuousFileReaderOperator.

Motivation: allow to avoid explicit synchronization and eventually to remove StreamTask.getCheckpointLock().

ContinuousFileReaderOperator was changed significantly to avoid
the currently necessary thread and extra synchronization between blocking and non-blocking code
(e.g.  format.reachedEnd and nextRecord)

Performance is lower by ~15% because of the overhead of enqueueing and processing each mail (instead of just loop).

zentol committed:
[hotfix][build][sql] Remove slf4j/log4j dependencies

The SQL client does not require explicit slf4j/log4j dependencies.
slf4j-api is already defined in the root pom, whereas the log4j-related dependencies are just not necessary since the distribution provides them.

zentol committed:
[hotfix][tests] Add "javax.management.*" to PowerMock ignore list

The test otherwise logged a LinkageError since it tries to load the MBeanServer interface which was already loaded.

zentol committed:
[hotfix][build] Add depMgmt entry for flink-shaded-hadoop

Some noise reduction.

twalthr committed:
[hotfix][table] Convert FlinkCalciteSqlValidator to Java

Converts the SqlValidator extension to Java. This class is likely
to extend Calcite. So it makes sense to have it in Java to quickly
copy paste code.

rmetzger committed:
[hotfix][e2e] Add retry timeout to kafka broker startup

rmetzger committed:
[hotfix][build system] Improve logging in maven setup script

rmetzger committed:
[hotfix][travis] Improve Maven dependency download stability

rmetzger committed:
[hotfix][travis] Use echo instead of printf

Azure exhibits some problems when using printf.

twalthr committed:
[hotfix][table] Simplify FlinkCalciteSqlValidator

This closes #10981.

azagrebin committed:
[FLINK-15143][docs] Add new memory configuration guide for FLIP-49

azagrebin committed:
[FLINK-15143][docs] Add tuning and troubleshooting guides for memory configuration

azagrebin committed:
[FLINK-15143][docs] Add migration guide from pre-FLIP-49 memory config

zentol committed:
[FLINK-15835][build][oss] Exclude hadoop-common

KarmaGYZ committed:
[FLINK-15417][e2e] Change the access permission of files modified in mesos constainer

This closes #10746.

rkhachatryan committed:
[FLINK-15811][task] report CancelTaskException on SourceStreamTask cancellation

Thread interruption and InterruptedException are implementation details
and shouldn't be exposed. Throw CancelTaskException instead which is a part of informal contract.

GJL committed:
[FLINK-15743][docs] Extend Flink 1.10 release notes

This closes #10997.

LululuAlu committed:
[FLINK-15866] Make ClosureCleaner "work" for classes that have no super class or interfaces

dianfu committed:
[FLINK-15897][python] Defer the deserialization of the Python UDF execution results

tillrohrmann committed:
[FLINK-15868] Pin jackson-dataformat-smile dependency to 2.10.1

tillrohrmann committed:
[FLINK-15868] Pin jackson-dataformat-yaml dependency to 2.10.1

zentol committed:
[hotfix][es][tests] Remove unused variable

Subsumed by #getUserConfig()

zentol committed:
[hotfix][es][tests] Refactor utils

zentol committed:
[FLINK-15868][es] Add tests for different formats

tillrohrmann committed:
[FLINK-15868] Bump flink-table-planner jackson dependencies to 2.10.1

tillrohrmann committed:
[FLINK-15868] Pin snakeyaml dependency in flink-connector-elasticsearch5 to 1.25

This closes #11006.

StephanEwen committed:
[hotfix][docs] Improve description of 'high-availability.jobmanager.port' config option.

StephanEwen committed:
[hotfix][runtime] Clean up minor issues in TaskInformation

  - Checkstyle violations in JavaDocs
  - Remove use of 'checkNotNull' on primitive types, which looks like a confusion/oversight.
    It is not checking anything and results in unnecessary boxing/unboxing.

StephanEwen committed:
[FLINK-15920][build] Show thread names in logs by default

This closes #11023

sunjincheng121 committed:
[FLINK-15921] [python] Add version range of grpcio. (#11024)

StephanEwen committed:
[hotfix][runtime] Small improvements in log messages for Task and RocksDB Backend

StephanEwen committed:
[hotfix][runtime] Small docs/comments cleanups in RPC code.

StephanEwen committed:
[FLINK-15966][runtime] Capture callstacks for RPC ask() calls to improve exceptions.

dianfu committed:
[FLINK-15902][python][docs] Improve the Python API documentation adding the version an API was introduced

This closes #11035

dianfu committed:
[FLINK-15929][python] Update the version limit of grpcio to 1.26.0

This closes #11027

sunjincheng121 committed:
[FLINK-15937][python] Update the Development Status to 5 - Production/Stable (#11028)

lirui-apache committed:
[FLINK-15933][catalog][doc] Update content of how generic table schema is stored in hive via HiveCatalog

closes #11029

wuchong committed:
[FLINK-15935][table] Fix watermark can't work when depending both on flink planner and blink planner in project

The reason is that we didn't sync the fixing of org.apache.calcite.sql.validate.ParameterScope to flink-planner. When an application project depends flink-planner and blink-planner at the same time, the classloader may use the ParameterScope from Calcite instead of from blink planner, which leads to this exception.

This closes #11030

wuchong committed:
[FLINK-15935][example] Add Streaming Window SQL example

zhijiangW committed:
[hotfix][task] Fix the comment typo in TwoInputStreamTask class

carp84 committed:
[FLINK-15919][core][mem] MemoryManager shouldn't allow releasing more memory than reserved

This closes #11025.

zentol committed:
[hotfix][build][e2e] Invert Java 11 activation condition

If we are not running on Java 11, then tests failing on Java 11 should still be run. Conversely, these tests should be excluded on Java 11.

zentol committed:
[hotfix][build][e2e] Remove empty include/exclude property

They shadow the properties set by the groovy script.

zentol committed:
[hotfix][build][e2e][addendum] Remove empty include/exclude property

twalthr committed:
[FLINK-15802][table] Bring input/output types of table functions closer

Refactors the code generation around table functions such that return data type
and argument data type are read from the function at the same location instead
of reading each in a separate code generator. This adds an additional runtime
collector that is only responsible for result conversion.

twalthr committed:
[FLINK-15802][table] Support new type inference for table functions

Further develops the stack based on BridgingSqlFunction and the new
type inference. Similar to FLINK-15487, we don't test every data type
combination yet.

This closes #11034.

yanghua committed:
[FLINK-15618][runtime] Remove unused JobTimeoutException

Commit:
[FLINK-14270][web] Support to display more metrics at once (#10689)

zentol committed:
[hotfix][travis] Fix caching of misc job

Adding the e2e-pre-commit profile activation only to the misc profile broke the caching

zentol committed:
[FLINK-15949][build] Harden jackson dependency constraints

wuchong committed:
[hotfix][example] Add StreamWindowSQLExample.jar to dist (#11036)

azagrebin committed:
[FLINK-15942] Do not log huge/infinite cpu/memory resources in profiles

GJL committed:
[FLINK-15917][runtime] Fix that the root exception is not shown in Web UI

GJL committed:
[FLINK-15918][runtime] Fix that 'uptime' metric is not reset after restart

The 'uptime' metric is the time difference between 'now' and the timestamp when
the job transitioned to state RUNNING. The new scheduler until now never
transitioned out of status RUNNING when restarting tasks which had the effect
that the 'uptime' metric was not reset after a restart. This introduces new
state transitions to the job. We transition the job status to RESTARTING if at
least one ExecutionVertex is waiting to be restarted, and we transition from
RESTARTING immediately to RUNNING again after the restart.

This closes #11032.

tillrohrmann committed:
[hotfix] Let ResourceProfileTest extend TestLogger

tillrohrmann committed:
[FLINK-15942][tests] Simplify ResourceProfileTest tests

This closes #11039.

yangjf2019 committed:
[hotfix][docs]  Fix some typos in state_processor_api docs (#11033)

lonerzzz committed:
[FLINK-15744] Log erroneous Task state changes at warn level

zentol committed:
[hotfix][travis] Add missing jdk11 classifier

wangxiyuan committed:
[hotfix][travis] Reduce depth for walkthrough module detection

If the module was built beforehand a pom.xml would be detected in the target directory.

HuangXingBo committed:
[FLINK-15913][python] Add Python TableFunction Runner and Operator in Blink planner

This closes #11044.

HuangXingBo committed:
[FLINK-15913][python] Add Python TableFunction Runner and Operator in old planner (#11020)

zhijiangW committed:
[FLINK-15914][checkpointing][metrics] Miss the checkpoint related metrics for the case of two inputs

When the StreamTwoInputSelectableProcessor was introduced before, it forgot adding the checkpoint related metrics in the constructor.
But it did not cause any problems, because only the StreamTwoInputProcessor actually worked before.

After StreamTwoInputProcessor is replaced by StreamTwoInputSelectableProcessor as now, this bug is exposed and we will not see the
checkpoint related metrics for the case of two inputs.

The solution is to add these metrics while constructing the CheckpointBarrierHandler.

JingsongLi committed:
[FLINK-15912][table] Add Context to TableSourceFactory and TableSinkFactory

This closes #11047

JingsongLi committed:
[FLINK-15912][table-planner-blink] Support create table source/sink by context in blink planner

curcur committed:
[hotfix][docs] Fix Readme Description for the State Machine Example (#11046)

zentol committed:
[hotfix][e2e] Reduce e2e-hadoop to exclusion

Also having it work as an include is confusing.

HuangXingBo committed:
[FLINK-15961][table-planner][table-planner-blink] Add physical Python Correlate rules and RelNodes

This closes #11051.

klion26 committed:
[FLINK-15741][docs-zh][TTL] Fix TTL docs after enabling RocksDB compaction filter by default

This closes #11022.

tisonkun committed:
[hotfix] Remove deadcode in ExecutionGraph#restart

tisonkun committed:
Revert "[hotfix] Remove deadcode in ExecutionGraph#restart"

This reverts commit 5f38f91a602201c4f52ac8febe13f9d72367a8c5.

dianfu committed:
[FLINK-15875][python] Bump Beam to 2.19.0

This closes #11050.

dianfu committed:
[FLINK-15875][python] Remove the ProcessManager which is copied from Beam 2.19

wuchong committed:
[hotfix][example] Copy StreamWindowSQLExample.jar to dist/examples/table

docete committed:
[FLINK-15546][table-planner-blink] Fix obscure error message from ScalarOperatorGens::generateCast (#11021)

JingsongLi committed:
[FLINK-15912][table-planner] Support create table source/sink by context in legacy planner

JingsongLi committed:
[FLINK-15912][table] Support create table source/sink by context in sql-cli

sunhaibotb committed:
[FLINK-14231][runtime] Introduce and use ProcessingTimeServiceAware to pass ProcessingTimeService to operator

JingsongLi committed:
[FLINK-15912][table] Support create table source/sink by context in hive connector

aljoscha committed:
[FLINK-15993] Move 404 "refresh" redirect into noscript block, add timeout

Apparently, search engines penalize META refresh links and this gets
around it

zhijiangW committed:
[FLINK-15914][tests] Fix the fragile tests caused by race condition of multiple threads

aljoscha committed:
[FLINK-15997] Make documentation 404 page look like a documentation page

carp84 committed:
[hotfix][docs] Add 1.10 to list of previous docs

rmetzger committed:
[hotfix][e2e] Fix docker invocation in mesos tests

rmetzger committed:
[hotfix][e2e] harden elasticsearch test

aljoscha committed:
[FLINK-16041] Expand "popular" documentation sections by default

aljoscha committed:
[minor][docs] Fix missing "page" text in redirect.html layout

aljoscha committed:
[FLINK-16000] Move "Project Build Setup" to "Getting Started" in documentation

We also add redirects.

dawidwys committed:
[FLINK-15862][kafka] Remove deprecated KafkaPartitioner

sunhaibotb committed:
[FLINK-14231][task] Add StreamOperatorWrapper that handles the close, endInput and other related logic of an operator

For each operator in the operator chain, its inputs must end completely before executing the "endInput()" method. For
the operator chain in a task, such as "OP1 - > OP2 - > ...", after the (source/network) input of OP1 are finished,
the operators on the chain are closed in the following order:

1. quiesce ProcessingTimeService of OP1 to prevent the pending timers from firing, but wait the timers in running
   to finish.
2. call OP1#close()
3. call OP2#endInput()
4. quiesce ProcessingTimeService of OP2 to prevent the pending timers from firing, but wait the timers in running
   to finish.
5. call OP2#close()
...

sunhaibotb committed:
[FLINK-14231][task] Change StreamTask to close operators with StreamOperatorWrapper to make the endInput semantics on the chain strict

sunhaibotb committed:
[hotfix][test] Clean up the codes in StreamTaskTestHarness, StreamTaskTimerTest and TestProcessingTimeServiceTest

These cleanups include removing redundant initializer, unnecessary type parameter declarations and redundant
suppression, and so on.

wangxlong committed:
[FLINK-15995][table-planner-blink] Change TO_BASE64 operand type checker from ANY to STRING

This can make the exception message more readable which can be thrown by Calcite instead of code generator. 

This closes #11058

rmetzger committed:
[FLINK-15982][e2e] Pick the right Flink version to fix the java quickstart test

This closes #11067

swhelan091 committed:
[FLINK-15941] [format-avro] Create schema coder only once in RegistryAvroSerializationSchema

wangxlong committed:
[FLINK-15994][table-planner-blink]Support byte array argument for FROM_BASE64 function (#11060)

docete committed:
[FLINK-15445][jdbc] Support new type system and check unsupported data types for JDBC table source

This bridges the JDBC table source to new type system to support precisions of types. In the meanwhile, this also adds a validation to check unsupported types and precisions by various dialects to avoid surprising results.

This closes #10745

JingsongLi committed:
[FLINK-15990][table][python] Remove register source and sink in ConnectTableDescriptor

This closes #11055

aljoscha committed:
[FLINK-15999][doc] Add outline of new concepts section

aljoscha committed:
[FLINK-15999][doc] Move explanation of DAGs/Dataflows from programming-model.md to stream-processing.md

JingsongLi committed:
[FLINK-15912][table] Clean TableFactoryUtil

aljoscha committed:
[FLINK-15999][doc] Clean up DAG/parallel DAG section in concepts doc

aljoscha committed:
[FLINK-15999][doc] Move state introduction from programming-model.md to stateful-stream-processing.md and clean up

aljoscha committed:
[FLINK-15999][doc] Move checkpointing introduction from programming-model.md to stateful-stream-processing.md

aljoscha committed:
[FLINK-15999][doc] Move batch-fault-tolerance from programming-model.md to stateful-stream-processing.md

zentol committed:
[hotfix][build] Add safeguards against rebranded jaxb-api/javax.activation-api dependencies

With jakarta EE having made their first releases projects may start transitioning to the re-branded versions.
The package names are identical (for now?), so we should be fine as long as we ensure that at least one version is on the classpath.

For the time being we stay on the "original" dependencies, but we may have to switch eventually.

jiasheng55 committed:
[FLINK-15992][kafka][es] Use thread context classloader when finding TableFormatFactory (#11064)

aljoscha committed:
[FLINK-15999][doc] Remove distinction between managed and raw state from state.md

Users should not be using raw state, except in rare edge cases. We can
just refer to managed keyed state ad keyed state and managed operator
state as operator state from now on.

aljoscha committed:
[FLINK-15999][doc] Move keyed state/operator state explanation from api doc to concepts

This also reorders/massages the text a bit to make it fit in better.

rkhachatryan committed:
[FLINK-16019][runtime] fix ContinuousFileReaderOperator error handling

aljoscha committed:
[FLINK-15999][doc] Move broadcast state explanation from api doc to concepts

This also reorders/massages the text a bit to make it fit in better.

zentol committed:
[FLINK-16004][build] Use correct parent

aljoscha committed:
[FLINK-15999][doc] Move checkpointing doc from internals to concepts

This is a straight copy without any adjustments. Follow-up commits will
massage this and make sure it fits well within the concepts section.

aljoscha committed:
[FLINK-15999][doc] Update newly moved checkpointing concepts section

This fixes links and merges the text with the previously existing text.
We also completely remove the section about actual snapshot
implementations in operators.

aljoscha committed:
[FLINK-15999][doc] Add TODOs to state concepts section

yanghua committed:
[hotfix][javadocs] Add missing asterisk

HuangXingBo committed:
[FLINK-16026][python] Limit the version of avro-python3

This closes #11078.

HuangXingBo committed:
[FLINK-15970][python] Optimize the Python UDF execution to only serialize the value (#11059)

zentol committed:
[hotfix][build] Downgrade groovy to 2.5.8

The google mirror for some reason does not contain 2.5.9.

zentol committed:
[hotfix][build] Remove unused flink-test-utils dependencies

zentol committed:
[hotfix][build] Remove redundant flink-test-utils-junit version/scope

aljoscha committed:
[FLINK-15999][doc] Move time and windows section from programming-model.md to timely-stream-processing.md

aljoscha committed:
[FLINK-15999][doc] Rename time concepts section to "Timely Stream Processing"

aljoscha committed:
[FLINK-15999][doc] Fix formatting and links in timely-stream-processing.md

aljoscha committed:
[FLINK-15999][doc] Move relevant sections from event_time.md to timely-stream-processing.md

This moves the conceptual parts from the development section to the
concepts section. massages them to fit in well and fixes formatting and
links.

aljoscha committed:
[FLINK-15999][doc] Add reference to windowing and ProcessFunction doc in event_time.md

Also fix section formatting.

aljoscha committed:
[FLINK-15999][doc] Remove "Low-level Operations" from ProcessFunction documentation

aljoscha committed:
[FLINK-15999][doc] Add TODOs in timely-stream-processing.md

aljoscha committed:
[minor] Remove outdated components(.zh).md

zentol committed:
[FLINK-16037][build] Bump dependency-analyzer to 1.11.1

pnowojski committed:
[hotfix][docs] Add note SSL not enabled by default

aljoscha committed:
[FLINK-15999][doc] Move state backend and savepoint doc from runtime.md to stateful-stream-processing.md

Also reformat/fix links and add TODOs.

aljoscha committed:
[FLINK-15999][doc] Move flink components doc from runtime.md to flink-architecture.md

Also rename and reformat to get up-to-date with current naming.

aljoscha committed:
[FLINK-15999][doc] Move remaining text from runtime.md to flink-architecture.md

Also rename and reformat to get up-to-date with current naming.

zentol committed:
[hotfix][web][build] Cleanup dependencies

zentol committed:
[hotfix][build] Remove unused shaded-asm7 dependencies

zentol committed:
[hotfix][runtime][tests] Use javax annotation

zentol committed:
[hotfix][core][tests] Mark SerializerTestInstance with @Ignore

Some runners may attempt to run this class as a test.

aljoscha committed:
[FLINK-16049] Remove outdated "Best Practices" section from Application Development Section

zentol committed:
[hotfix][build] Remove unused akka-testkit dependencies

StephanEwen committed:
[FLINK-10918][state backends] Change o.a.f.core.fs.Path to java.nio.file.Path in RocksDB incremental checkpoints

This solves the bug that Paths are handled wrong on Windows, breaking incremental checkpoints.

This closs #11095

HuangXingBo committed:
[FLINK-16040][python] Change local import to global import (#11084)

zentol committed:
[FLINK-16046][es] Drop Elasticsearch 2 connector

rkhachatryan committed:
[FLINK-16057][runtime] chain ContinuousFileReaderOperator by default (HEAD)

dianfu committed:
[FLINK-16053][python] Remove redundant metrics in PyFlink (#11089)

lirui-apache committed:
[FLINK-16055][hive] Avoid catalog functions when listing Hive built-in functions

closes #11093

rkhachatryan committed:
[FLINK-16056][runtime][tests] do fail ContinuousFileProcessingITCase on failure

zentol committed:
[FLINK-16169][tests] Harden BlobServerRangeTest

The test is unreliable since there is no guarantee that the "availablePort" is still available then the blob server starts.

twalthr committed:
[hotfix][table-common] Remove deprecated DataTypes.ANY(...)

This closes #11094.

zentol committed:
[hotfix][build] Remove various unused test dependencies

rkhachatryan committed:
[FLINK-16056][runtime][tests] create ContinuousFileReaderOperator using factory only
Also drop test constructor and make the class package-private.

zentol committed:
[FLINK-16059][rest][docs] Order properties alphabetically

StephanEwen committed:
[FLINK-16065][core] Unignore FileUtilsTest.testDeleteDirectoryConcurrently()

StephanEwen committed:
[hotfix][tests] Adjust FileUtilsTest.testDeleteSymbolicLinkDirectory() to handle unsupported situations in Windows

StephanEwen committed:
[hotfix][core] Fix broken JavaDoc link in FileUtils

zentol committed:
[hotfix][docs][conf] Add query service port to port section

zentol committed:
[hotfix][docs][conf] Add query service port to port section

zentol committed:
[hotfix][docs][conf] Setup logging for generator

zentol committed:
[hotfix][docs][conf] Log included packages / excluded classes

Eases discoverability

wuchong committed:
[docs-sync] Synchronize the latest documentation changes into Chinese documents

rmetzger committed:
[FLINK-13978][build system] Add experimental support for building on Azure Pipelines

This closes #10976

rmetzger committed:
[hotfix][build system] broader search for YARN logs

rmetzger committed:
[hotfix] Harden kubernetes test

Commit:
[FLINK-15702][sql-client] Respect CLASSLOADER_RESOLVE_ORDER in sql-client

This closes #10910 .

HuangXingBo committed:
[hotfix][docs] Regenerate documentation

yanghua committed:
[FLINK-15708] Add MigrationVersion.v1_10

HuangXingBo committed:
[FLINK-16109][python] Move the Python scalar operators and table operators to separate package

aljoscha committed:
[FLINK-16000] Add "where to go" to getting-started overview section

This also re-orders the overview sections to make the ordering
consistent between the overview and the actual sub-sections.

aljoscha committed:
[FLINK-16000] Remove icons from getting-started subsections

This makes the section more consistent with the rest of the
documentation, where no sub-sections have icons.

aljoscha committed:
[FLINK-15999][doc] Remove "Asynchronous State Snapshots" section from concepts

aljoscha committed:
[FLINK-15999][doc] Remove "ingestion time" from concepts section

Ingestion time is a special "flavour" of event time and one that might
not be too useful in practice because of inconsistencies that arise in
case of failures/recovery. If users don't need event-time they can go
straight to processing time.

aljoscha committed:
[FLINK-15999][doc] Remove "operator state" from concepts section

We move it to the development section and mention that it is a special
type of state that is normally not needed in user programs.

aljoscha committed:
[FLINK-15999][doc] Remove programming-model section and incorporate into overview

This also adds explanation about the concepts section

twalthr committed:
[FLINK-16118][table-planner-blink] Ignore order for FunctionITCase.testDynamicTableFunction

aljoscha committed:
[FLINK-15999][doc] Update/remove outdated information in concepts section

The content of the concepts section was gathered from pre-existing
documentation texts. This commit cleans that up a bit, removes outdated
content that is not "conceptual" and updates some of the terms that we
no longer use.

HuangXingBo committed:
[FLINK-16071][python] Introduce FlattenRowCoder to solve the performance issue of __get_item__ in Row

wuchong committed:
[FLINK-16113][table-planner-blink] ExpressionReducer shouldn't escape the reduced string value (#11108)

StephanEwen committed:
[hotfix][config] Adjust AkkaOptions to new type-safe pattern

libenchao committed:
[FLINK-16068][table-planner-blink] Fix DDL fails when both computed column and keyword-escaped column exist (#11101)

libenchao committed:
[FLINK-15988][json] Make JsonRowSerializationSchema's constructor private (#11080)

StephanEwen committed:
[FLINK-14086][core] (follow-up) Clean-up / unify processor and memory architecture enums

This closes #9768

StephanEwen committed:
[FLINK-15473][core] Add ppc64le to the list of known processor architectures.

StephanEwen committed:
[FLINK-16135][core] Unify the TemporaryClassLoaderContext and AutoContextClassLoader

These two classes did teh same thing, this commit unifies the use to a consolidated instance.

This closes #11120

rmetzger committed:
[FLINK-16136][yarn][tests] Increase disk space limit for all YARN tests

zhuzhurk committed:
[FLINK-16139][runtime] Reset colocation constraints when restarting tasks in DefaultScheduler

rkhachatryan committed:
[FLINK-16056][docs] update StreamExecutionEnvironment javadoc

rmetzger committed:
[FLINK-15833] Fix 'Kerberized YARN on Docker' e2e test on Azure Pipelines

Mostly removed the "-it" argument from docker exec. -it is meant to be used for interactive shell sessions, not in scripts on a CI system.
Because of the "-it" argument, the docker command printed "the input device is not a TTY" to the output. That's why some checks / assumptions in the scripts failed.

aljoscha committed:
[FLINK-11589] Add tests for SecurityContext fallback behaviour

zentol committed:
[FLINK-16162][build] Bump flink-shaded to 10.0

zentol committed:
[FLINK-16163][build] Migrate to flink-shaded-zookeeper

Commit:
[FLINK-15904][connectors/kafka] Use explicit Serializer for KafkaConsumer unionOffsetStates

walterddr committed:
[FLINK-11589] Regenerate security configuration docs

YuvalItzchakov committed:
[FLINK-14533][flink-table-planner] Fix LOWER/UPPER not being pushed down to TableSource (#10005)

dawidwys committed:
[FLINK-16033][table-api] Construct expressions through factory methods (#11141)

This commit is a preparation step for next commits. The only entry point
to constructing api expression are methods in ApiExpressionUtils. This
way we can put additional logic in the factory methods without polluting
the ctors.

zentol committed:
[FLINK-16164][build] Disable maven-site-plugin

dawidwys committed:
[FLINK-16182][table-api] Remove check against null types as a result of an input type inference when deriving output type based on surrounding info.

tisonkun committed:
[hotfix] Refactor YARNFileReplicationITCase to avoid inheritance

dawidwys committed:
[hotfix][table-api] Fixed ExpressionResolver rules modifiers.

zentol committed:
[FLINK-16186][es][tests] Reduce connect timeout to 5 seconds

HuangXingBo committed:
[FLINK-16072][python] Optimize the performance of the write/read null mask of FlattenRowCoder

pnowojski committed:
[FLINK-16019][runtime][test] Implement test coverage for FLINK-16019 bug

StephanEwen committed:
[FLINK-16191][state backends] Improve error message on Windows when RocksDB is give a too long path.

rmetzger committed:
[hotfix][build system] Do not trigger dependent jobs if predecessor fails

StephanEwen committed:
[hotfix][DataStream API] Minor JavaDoc / deprecation cleanup in CheckpointedFunction

tillrohrmann committed:
[FLINK-14038][docs] Extend Application Profiling documentation

This commit adds explanations on how to analyze/debug out of memory problems and
how to analyze GC behaviour.

This closes #11165.

tisonkun committed:
[FLINK-15652][kubernetes] Change imagePullSecrets option to list ConfigOption

This closes #11166 .

wuchong committed:
[hotfix] Revert "[FLINK-15992][kafka][es] Use thread context classloader when finding TableFormatFactory (#11064)"

This reverts commit c6158648

tisonkun committed:
[FLINK-15977][docs] Update pull request template to include Kubernetes as deployment candidates

This closes #11053 .

rmetzger committed:
[hotfix] Stabilize python tests

tisonkun committed:
[FLINK-15967][examples] Use universal kafka connector in StateMachineExample

afedulov committed:
[FLINK-16222][tests] Prepare e2e tests

- Add functionality to add JARs to plugin directory
- Add metrics reporter initialization tests for plugins mechanism (using
both reflection and factory mechanisms) and a combined libs + plugins
initialization

afedulov committed:
[FLINK-16222][core] Relax Plugin bounded type parameter constraint

afedulov committed:
[FLINK-16222][metrics] Factorify and pluginize Prometheus reporter

afedulov committed:
[FLINK-16222][runtime] Use plugins load mechanism to initialize MetricsReporters

StephanEwen committed:
[hotfix][scripts] Fix comment for default GC configuration

libenchao committed:
[FLINK-16082][docs-zh][table] Translate "Streaming Concepts Overview" page into Chinese (#11128)

zentol committed:
[FLINK-16233][hive][build] Add additional log4j 1 exclusions

Adds several missing log4j1 exclusions, that are only necessary for certain hive versions.

pnowojski commented:
Superseded by https://github.com/apache/flink/pull/11195
